object_manager.go

```diff
@@ -376,12 +410,14 @@ func (objMgr *ObjectManager) CreateEADefinition(eadef EADefinition) (*EADefiniti
 	return newEadef, err
 }
 
-func (objMgr *ObjectManager) CreateHostRecordWithoutDNS(recordName string, netview string, cidr string, ipAddr string, macAddress string, vmID string) (*HostRecord, error) {
+func (objMgr *ObjectManager) CreateHostRecord(enabledns bool, recordName string, netview string, dnsview string, cidr string, ipAddr string, macAddress string, vmID string, vmName string) (*HostRecord, error) {
```

Comment:
You can split this into multiple functions rather than just having `CreateHostRecord`. This will have more if-else conditions as we find more cases. You can do this

Make `createHostRecord` generic which takes `extrattrs` etc.

Then have 
`CreateHostRecordWithoutDNS`
`CreateHostRecordForIp`
`CreateHostRecordFromRange` and so on which will call `createHostRecord`

Reply:
actually the enableDns flag here decides if we need to create a host Record for DNS purposes or not.
It is like this , if we pass true for enable dns then we will have a host record created under that particular view and if we dont then it will just create a hostrecord for IPAM purposes